File: HCrawler.Api/Controllers/ImagesController.cs
using System.Threading.Tasks;
using HCrawler.Api.ViewModels;
using HCrawler.Core;
using Microsoft.AspNetCore.Mvc;

namespace HCrawler.Api.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class ImagesController : ControllerBase
    {
        private readonly Image.Image _image;

        public ImagesController(Image.Image image)
        {
            _image = image;
        }

        [HttpGet]
        public async Task<IActionResult> Get([FromQuery] PageFilter filter)
        {
            var page = await _image.getAllAsync(filter.ToRecord());
            return Ok(page);
        }

        [HttpPost]
        public async Task<IActionResult> Post([FromBody] CreateImage createImage)
        {
            await _image.createImageIfNotExistsAsync(createImage.ToRecord());
            return Ok();
        }
    }
}

File: HCrawler.Api/Controllers/ImagesController.fs
namespace HCrawler.Api.Controllers

open HCrawler.Core.Image
open HCrawler.Core.Payloads
open Microsoft.AspNetCore.Mvc
open FSharp.Control.Tasks.V2

[<ApiController>]
[<Route("[controller]")>]
type ImagesController(image: Image) =
    inherit ControllerBase()

    [<HttpGet>]
    member __.Get([<FromQuery>] filter) = image.getAllAsync filter
    
    [<HttpPost>]
    member __.Post([<FromBody>] createImage) = image.createImageAsync createImage
